docs/hello_nextflow/02_hello_channels.md

```diff
@@ -0,0 +1,868 @@
+```
+
+Running this should feel extremely familiar by now.
+
+```console title="Output"
+ N E X T F L O W   ~  version 24.10.0
+
+ ┃ Launching `hello-channels.nf` [cheesy_engelbart] DSL2 - revision: b58b6ab94b
+
+executor >  local (1)
+[1c/9b6dc9] sayHello (1) [100%] 1 of 1 ✔
+```
+
+Be sure to open up the output file to check that you now have the new version of the greeting. Voilà!
+
+!!! tip
+
+    You can readily distinguish Nextflow-level parameters from pipeline-level parameters.
+
+    - Parameters that apply to a pipeline always take a double hyphen (`--`).
+    - Parameters that modify a Nextflow setting, _e.g._ the `-resume` feature we used earlier, take a single hyphen (`-`).
+
+### 3.3. Set a default value for a command line parameter
+
+In many cases, it makes sense to supply a default value for a given parameter so that you don't have to specify it for every run.
+
+Let's initialize the `greeting` parameter with a default value by adding a parameter declaration before the workflow definition.
+
+```groovy title="hello-channels.nf" linenums="22"
+/*
+ * Pipeline parameters
+ */
+params.greeting = 'Holà mundo!'
+```
+
+!!! tip
+
+    You can put the parameter declaration inside the workflow block if you prefer. Whatever you choose, try to group similar things in the same place so you don't end up with declarations all over the place.
+
+### 3.4. Run the workflow again without specifying the parameter
+
+Now that you have a default value set, you can run the workflow again without having to specify a value in the command line.
+
+```bash
+nextflow run hello-channels.nf
+```
+
+The console output should look the same.
+
+```console title="Output"
+ N E X T F L O W   ~  version 24.10.0
+
+ ┃ Launching `hello-channels.nf` [wise_waddington] DSL2 - revision: 988fc779cf
+
+executor >  local (1)
+[c0/8b8332] sayHello (1) [100%] 1 of 1 ✔
+```
+
+Check the output in the results directory, and... Tadaa! It works!
+Nextflow used the default value to name the output.
+
+### 3.5. Run the workflow again with the parameter to override the default value
+
+If you provide the parameter on the command line, the CLI value will override the default value.
+
+Try it out:
+
+```bash
+nextflow run hello-channels.nf --greeting 'Konnichiwa!'
+```
+
+The console output should look the same, and you will have the corresponding new output in your results directory.
+
+!!! note
+
+    In Nextflow, there are multiple places where you can specify values for parameters.
+    If the same parameter is set to different values in multiple places, Nexflow will determine what value to use based on the order of precedence that is described [here](https://www.nextflow.io/docs/latest/config.html).
+
+### Takeaway
+
+You know how to use CLI parameters to feed inputs to the workflow.
+
+### What's next?
+
+Learn how to manage channel contents using operators.
+
+---
+
+## 4. Supply a batch of multiple values via the `params` system
```

Comment:
Would this not be a good place to stop, and have a subsequent (possibly quite short) 'hello operators', using `flatten` and `splitCsv` as exemplars?

Reply:
As discussed in slack, I'll add some more explicit text calling out the importance of operators in section 4 and won't split anything out for now.